### PR TITLE
Support multiple root cert provision

### PR DESCRIPTION
**Pull request by Wenxing-hou**
Fix: #151 
Support multiple root cert provision.

1. Change spdm_context->local_context.peer_root_cert_provision and spdm_context->local_context.peer_root_cert_provision_size to array.
The multiple root cert store in the array. The MAX_ROOT_CERT_SUPPORT is 10 by default.

2. Set data for local root cert:

- Iterate through the all root_cert_provision array in order.
- Find the first NULL root_cert_provision[i] to set root cert.
- When the all root_cert_provision is full, return RETURN_OUT_OF_RESOURCES.

3. verify local root cert and peer cert chain:

- Iterate through the all root_cert_provision array in order.
- Compare every root_cert_hash, until find the match root cert.
- otherwise, return FALSE.

4. Update the unit_test. Change root_cert_provision to root_cert_provision array.

5. Add new unit tests for spdm_verify_peer_cert_chain_buffer and libspdm_set_data.

6. Update user_guide.md to guide setting multi root certs

Signed-off-by: Wenxing Hou <wenxing.hou@intel.com>

### CODE REVIEW

File: include/industry_standard/spdm.h [hunk 1]
```diff
@@ -19,6 +19,7 @@
 #define SPDM_MAX_SLOT_COUNT 8
 #define SPDM_MAX_OPAQUE_DATA_SIZE 1024
 
+#define MAX_ROOT_CERT_SUPPORT 10
```
Comment (jyao1) [inline include/industry_standard/spdm.h:22]: please move this to config.h, and add LIBSPDM_ prefix.

**Comment by jyao1** [comment]
please add unit test.

File: include/industry_standard/spdm.h (hunk 1 above)
Comment (Wenxing-hou) [reply include/industry_standard/spdm.h]: Yes, I have changed it. Thanks.

**Comment by jyao1** [comment]
Looks good

**Review by steven-bellock — APPROVED** [review]

**Pull request merged**